src/App.tsx

```diff
@@ -1,25 +1,75 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { FilterTodos } from './types/FilterTodos';
+import { ErrorType } from './types/Errormessage';
+import { getTodos } from './api/todos';
+import { TodoFilter } from './components/TodoFilter';
+
+const USER_ID = 11500;
+
+function getFilteredTodos(
+  todos: Todo[],
+  sortBy: string,
+) {
+  switch (sortBy) {
+    case FilterTodos.All:
+      return todos;
+
+    case FilterTodos.Active:
+      return todos.filter(todo => todo.completed === false);
+
+    case FilterTodos.Completed:
+      return todos.filter(todo => todo.completed === true);
+
+    default:
+      return todos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<string>(FilterTodos.All);
+  const [errorMessage, setErrorMessage] = useState(ErrorType.None);
+  const [isErrorClosed, setIsErrorClosed] = useState(false);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorType.LoadingError);
+        setIsErrorClosed(false);
+        setTimeout(() => {
+          setIsErrorClosed(true);
+        }, 3000);
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = getFilteredTodos(todos, filterTodos);
+  const activeTodos = getFilteredTodos(todos, FilterTodos.Active);
+  const completedTodos = getFilteredTodos(todos, FilterTodos.Completed);
```

Comment:
This is not optimal

It is better to calculate only on list of todos you should display, based on selected filters